src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -304,7 +304,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       </PropertyGroup>
     </When>
     
-    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(OutputType)' == 'Exe'">
+    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(IsExecutable)' == 'true'">
```

Comment:
Test projects aren't "executable".  You can't double-click on it's output and expect the test to run.  You can't type into a command-line `C:\mytest\mytest.dll` and watch the tests run.

> I am open for a better name.

The list I came up with before (which we can add more to) was:

- TreatAsExecutable
- HasRuntimeOutput
- HasRunnableOutput
- HasExecutableOutput
- IsExecutable
- IsRunnable

I think the one I like most, given all the situations this is going to be used, is what we landed on in the project.json world: `HasRuntimeOutput`.  The output isn't "runnable" or "executable", but instead the output is associated with a runtime, and that's why we can generate binding redirects and deps.json files that will work at runtime.

> I would think this would be valid and part of setting IsExecutable=true would involve having uses configure the F5 information (to use xunit, nunit, dotnet, etc...)

That's exactly my point.  Look at the rest of the code around this area that I commented on.  If `IsExecutable=true` on a test project, the "Run Information" being returned says that you can just run the output directly:

```XML
<PropertyGroup>
  <RunCommand Condition="'$(RunCommand)' == ''">$(TargetPath)</RunCommand>
  <RunArguments Condition="'$(RunArguments)' == ''">$(StartArguments)</RunArguments>
```

> However, maybe having a separate variable to disable (or explicitly enable) F5 for some scenarios would be desirable

Again, that's what the `RunCommand` property is for.  If it is set, you can `dotnet run` or F5 this project.  If it isn't set, then you can't.